### PR TITLE
My Listings Page

### LINKED ISSUE #74: 𝗣𝗮𝗴𝗲: Edit/Manage Listing Page
## Feature Description

**Create/Manage Listings: Users should be able to create new listings, and be able to edit, delete existing listings**
This page is very similar to #73(Create Listing), and it requires all of the same fields but has the addtion of having a "Remove Posting" button, a "Mark Purchased/Unpurchased" button, and a "Cancel" button which would bring you back to your account page. Additionally clicking the "Update Listing" button sends a patch request to update a listing.  This requires components of:
- #42 
- #45 

relates: #42, #45, #73 
blocked by: #42, #45

## Problem You're Trying to Solve

See [Manage a Lisiting](https://docs.google.com/document/d/1DNqexMBf4MpYPCxWsT3plUDAjDSbdJwUzw-UyZX-vDs/edit#heading=h.w6c7eks31w0j) and [Delete a Listing](https://docs.google.com/document/d/1DNqexMBf4MpYPCxWsT3plUDAjDSbdJwUzw-UyZX-vDs/edit#heading=h.bx2hjhcsjeky) for User Stories

## Related External Resources

[Edit Listing see OpenAPI](http://market.lebl.ca/openapi/#/Listings/editListing)

Image

### PR DESCRIPTION
**Pull request by sejal-agarwal**
# Description

Screenshot 2024-06-24 at 8 22 22 PM

Closes #74

## Checklist
- [ ] The code includes tests if relevant
- [ ] I have *actually* self-reviewed my changes and done QA

### CODE REVIEW

**Review by CaelumD25 — APPROVED** [review]
There seem to be some minor graphical issues, but other than that it seems to be good! At least, I can't find anything critical at 10:45 at night.

File: apps/frontend/src/pages/MyListings.tsx [hunk 1]
```diff
@@ -1,25 +1,81 @@
...
+    >
       {isDesktop && <AccountSidebar selectedItem="My Listings" />}
-      <Box
-        sx={{
-          paddingLeft: isDesktop ? "400px" : "0",
-          display: "flex",
-          height: "100vh",
-        }}
+      {!isDesktop && <SearchBar />}
+      <Typography variant={"h4"}>My Listings</Typography>
```
Comment (Scott-Kenning) [inline apps/frontend/src/pages/MyListings.tsx:59]:
```suggestion
      <Typography variant="h4">My Listings</Typography>
```

File: apps/frontend/src/components/listingCard.tsx [hunk 2]
```diff
@@ -106,9 +116,9 @@ const ListingCard = ({ searchPerformed, listing }: ListingCardProps) => {
               textTransform: "none",
               width: "calc(100% - 20px)",
             }}
-            onClick={handleNotInterested}
+            onClick={canEdit ? handleCanEdit : handleNotInterested}
           >
-            Not interested
+            {canEdit ? "Edit" : "Not interested"}
```
Comment (Scott-Kenning) [inline apps/frontend/src/components/listingCard.tsx:121]: I think this is one of the times where more code is better, having an "Edit" vs "Not Interested" button might be more semantic. It also might be better for the user experience if the button for stopping suggestions on someone else's listing vs editing their own listing are different colours. Definitely doesn't need changing, but something to consider.

File: apps/frontend/src/pages/MyListings.tsx [hunk 3]
```diff
@@ -1,25 +1,81 @@
...
+      }}
+    >
       {isDesktop && <AccountSidebar selectedItem="My Listings" />}
-      <Box
-        sx={{
-          paddingLeft: isDesktop ? "400px" : "0",
-          display: "flex",
-          height: "100vh",
-        }}
+      {!isDesktop && <SearchBar />}
```
Comment (Scott-Kenning) [inline apps/frontend/src/pages/MyListings.tsx:58]: Is SearchBar the mobile equivalent to the account sidebar? if it is then I would suggest doing this to indicate that
`isDesktop ? <AccountSidebar /> : <SearchBar/>`

If not, then mobile or desktop view (or both) are going to be missing stuff

Comment (sejal-agarwal) [reply apps/frontend/src/pages/MyListings.tsx:58]: Yes, the Mobile <SearchBar/> includes AccountSidebar components :)

File: apps/frontend/src/components/listingCard.tsx (hunk 2 above)
Comment (sejal-agarwal) [reply apps/frontend/src/components/listingCard.tsx:121]: Two different buttons now. 
Screenshot 2024-06-25 at 7 56 13 AM

**Review by ChloeZacharias — APPROVED** [review]
Looks great other than some graphical things but I'm not sure how much of a priority those are to fix. it can also probably be fixed in a different PR if we have time

The account bar going over the search bar
image
image

The my listing title not looking centered
image

The images not being the same width as the edit button (This is probably something I missed in my PR)
image

and the search bar not going to the end of the page
image

**Pull request merged**